docs/go/extras/extras.md

```diff
@@ -4,7 +4,7 @@
   [gazelle rule]: https://github.com/bazelbuild/bazel-gazelle#bazel-rule
   [golang/mock]: https://github.com/golang/mock
   [gomock_rule]: https://github.com/jmhodges/bazel_gomock
-  [core go rules]: core.rst
+  [core go rules]: core.md
```

Comment:
I think this is now a broken link after moving files around?

I would rather have all the .md files in the docs/ folder. I don't think subdirectories are useful, just makes it harder for users to navigate around

Reply:
I did an audit of all the broken links, they should be fixed now. 

There are a few problems with keeping the .md files in a flat structure:
  - Collisions with section files (eg. rules.md for go/core.rst and rules.md for proto/core.rst)
  - Can't distinguish identically named files in different directories (eg. go/core.md vs proto/core.md) 

The only way to get around the above is with naming the files with some scheme (eg. go_core.md and proto_core.md), but this will get unwieldy. Moreover looking at our docsite, it looks like there are repos which have md files organized in subdirectories.